yt/geometry/coordinates/spherical_coordinates.py

```diff
@@ -158,6 +158,14 @@ def pixelize(
                 data_source, field, bounds, size, antialias, dimension, periodic
             )
         elif name in ("theta", "phi"):
+            if name == "theta":
+                # This is admittetly a very hacky way to resolve a bug
+                # it's very likely that the *right* fix would have to
+                # be applied upstream of this function, *but* this case
+                # has never worked properly so maybe it's still preferable to
+                # not having a solution ?
+                # see https://github.com/yt-project/yt/pull/3533
+                bounds = (*bounds[2:4], *bounds[:2])
```

Comment:
Your comment makes me think that we need to have a discussion about moving some of these functions around a bit.  It seems that you've likely already come up to speed on what everything does and would be a good person to speculate about potential refactorings that touch the coordinate handlers.  Could we set up a time to chat?

Reply:
Actually I went with a very dumb approach of touching everything in a sensible way until something worked. Now I wouldn't say I was able to understand most of it, and I don't think I would be in a position to champion such a refactor, but _yes_, let's set up a time to discuss that over zoom :)